spec/bigcommerce/prometheus/collectors/base_spec.rb

```diff
@@ -22,6 +22,18 @@
   let(:client) { double(:client, send_json: true) }
   let(:collector) { AppCollector.new(client: client, frequency: 0) }
 
+  describe '#start' do
+    subject { AppCollector.start(client: client, frequency: 0) }
+
+    it 'starts the collector on a thread' do
+      thread = subject
+      expect(thread).to be_a(Thread)
+      expect(thread).to be_alive
+    ensure
+      AppCollector.stop
```

Comment:
Could also be done as an `after` if you expect any other tests here.

Reply:
I preferred being explicit and not relying on rspec to catch the error (so that i didnt have a leaked thread).